flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -717,6 +717,18 @@ public boolean isOnlyConsumeFinishedPartition() {
                                                             code(UNFINISHED_PRODUCERS.name())))
                                             .build());
 
+    @Documentation.ExcludeFromDocumentation(
+            "this is an expert option, that we do not want to expose in the documentation")
+    public static final ConfigOption<Integer> OFFLOAD_SHUFFLE_DESCRIPTORS_THRESHOLD =
```

Comment:
I suggest to move this config option to `TaskDeploymentDescriptorFactory` and marked as `@Experimental`. We can put it back to `JobManagerOptions`  when we think it should be exposed to users